modules/placement-driver/src/test/java/org/apache/ignite/internal/placementdriver/LeaseUpdaterTest.java

```diff
@@ -117,19 +123,30 @@ public class LeaseUpdaterTest extends BaseIgniteAbstractTest {
     void setUp() {
         HybridClockImpl clock = new HybridClockImpl();
 
-        Entry entry = new EntryImpl(
+        Entry stableEntry = new EntryImpl(
                 stablePartAssignmentsKey(new TablePartitionId(1, 0)).bytes(),
                 Assignments.of(HybridTimestamp.MIN_VALUE.longValue(), Assignment.forPeer(node.name())).toBytes(),
                 1,
                 clock.now()
         );
 
-        when(mcEntriesCursor.iterator()).thenReturn(List.of(entry).iterator());
+        Entry pendingEntry = new EntryImpl(
+                pendingPartAssignmentsKey(new TablePartitionId(1, 0)).bytes(),
+                Assignments.of(HybridTimestamp.MIN_VALUE.longValue(), Assignment.forPeer(node.name())).toBytes(),
+                1,
+                clock.now()
+        );
+
+        when(msStableAssignmentsEntriesCursor.iterator()).thenReturn(List.of(stableEntry).iterator());
```

Comment:
As a unit test? Because after https://github.com/apache/ignite-3/pull/4446 I bet we have an integration test that models such situation (that's why we're fixing it).

Reply:
I would prefer to have it as a placement driver specific unit test. Integration test related to other modules can be changed anytime, but since we extend the possibilities of placement driver, the corresponding test should exist for regression purposes.

Reply:
Small new `testLeaseAmongPendings` test is added and thanks to it -- I found several dumb bugs :'>